### ISSUE TITLE
Dialogs broken in develop

### ISSUE THREAD

**Issue by JonnyTech** [graemeg/fpGUI#45]
Dialogs are broken in the develop branch - working in master.
Clicking on OK after showing a dialog displays (linux64):

**An unexpected error has occurred
ess violation**

Sample program:

```
program test;

{$mode objfpc}{$H+}

uses
  {$IFDEF UNIX}{$IFDEF UseCThreads}
  cthreads,
  {$ENDIF}{$ENDIF}
  Classes, SysUtils,
  fpg_base, fpg_main, fpg_form, fpg_dialogs, fpg_button;

type
  TMainForm = class(TfpgForm)
  private
    button:   TfpgButton;
    procedure   buttonClick(Sender: TObject);
  protected
  public
    constructor Create(AOwner: TComponent); override;
  end;

{ TMainForm }

procedure TMainForm.buttonClick(Sender: TObject);
begin
  ShowMessage('OK');
end;

constructor TMainForm.Create(AOwner: TComponent);
begin
  inherited Create(AOwner);
  Width := 200;
  Height := 120;
  button := CreateButton(self, 20, 20, 75, 'TEST', @buttonClick);
end;

procedure MainProc;
var
  frm: TMainForm;
begin
  fpgApplication.Initialize;
  frm := TMainForm.Create(nil);
  frm.Show;
  fpgApplication.Run;
  frm.Free;
end;

begin
  MainProc;
end.
```

**Comment by graemeg** [comment]
Thanks for reporting the issue. This bug should now be fixed as of commit e81712563. Please get an update and retest.

**Comment by JonnyTech** [comment]
Fix confirmed - thanks
